circle.yml

```diff
@@ -0,0 +1,9 @@
+machine:
+  services:
+    - docker
+
+test:
+    override:
+        - (chmod +x setup_environment.sh)
```

Comment:
Not necessary. I am assuming you work on Linux.
Before doing (git add + git commit) do locally on your PC: `chmod +x setup_environment.sh`
GIT keeps the file together with mode of the file.